Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -174,118 +169,160 @@ public IConnection Acquire(AccessMode mode)
             return Acquire();
         }
 
-        public Task<IConnection> AcquireAsync(AccessMode mode)
+        public IPooledConnection Acquire()
         {
-            return AcquireAsync();
+            using (var timeOutTokenSource = new CancellationTokenSource(_connAcquisitionTimeout))
+            {
+                return Acquire(timeOutTokenSource.Token);
+            }
         }
 
-        public IPooledConnection Acquire()
+        private IPooledConnection Acquire(CancellationToken cancellationToken)
         {
-            return TryExecute(() =>
+            try
             {
-                if (_disposeCalled)
+                return TryExecute(() =>
                 {
-                    ThrowObjectDisposedException();
-                }
-                IPooledConnection connection;
+                    IPooledConnection connection = null;
 
-                if (!_availableConnections.TryTake(out connection))
-                {
-                    var connAcquisitionTimer = new Stopwatch();
-                    connAcquisitionTimer.Start();
-                    do
+                    while (true)
                     {
-                        if (!IsConnectionPoolFull())
+                        if (_disposeCalled)
                         {
-                            connection = CreateNewPooledConnection();
-                            break;
+                            ThrowObjectDisposedException();
                         }
-                        if (_availableConnections.TryTake(out connection, ConnAcquisitionQueuingTimeout))
+
+                        cancellationToken.ThrowIfCancellationRequested();
+
+                        if (!_availableConnections.TryTake(out connection, 0, cancellationToken))
                         {
-                            break;
+                            do
+                            {
+                                if (!IsConnectionPoolFull())
+                                {
+                                    connection = CreateNewPooledConnection();
+                                    break;
+                                }
+
+                                if (_availableConnections.TryTake(out connection, 0, cancellationToken))
```

Comment:
Why removed `ConnAcquisitionQueuingTimeout`? We initially want this blocking queue is to make thread wait a bit on this `TryTake` method to introduce some queuing/waiting inside async. I am a bit worried if we return directly without any sleep, will the thread span here a lot?

Reply:
I've put a small time lapse back into the loop (not a magnitude of seconds but milliseconds) but remove those timeouts and cancellationtoken checking logic from the first TryTake - making the fast path do not bother with those.